hpe3parclient/client.py

```diff
@@ -494,8 +505,35 @@ def createVolume(self, name, cpgName, sizeMiB, optional=None):
 
         """
         info = {'name': name, 'cpg': cpgName, 'sizeMiB': sizeMiB}
+        # For merlin array there is no compression and tdvv fields
+        # removing tdvv and replacing compression with reduce field
+        if not optional and self.merlin_supported:
+            optional = {}
+            optional['tpvv'] = True
         if optional:
+            if self.merlin_supported:
+                if 'tdvv' in optional:
+                    optional.pop('tdvv')
+
+                if optional.get('tpvv') and optional.get('compression'):
+                    optional['reduce'] = True
+
+                if not optional.get('tpvv') \
+                        and not optional.get('compression'):
+                    optional['tpvv'] = True
+
+                if 'compression' in optional:
+                    if optional.get('compression'):
+                        optional['reduce'] = True
+                    else:
+                        optional['reduce'] = False
+
+                    optional.pop('compression')
+
+                if optional.get('tpvv') and optional.get('reduce'):
+                    optional.pop('tpvv')
```

Comment:
![PossibleInputsAndFinalResponse-TruthTable](https://user-images.githubusercontent.com/31840833/60660548-a2ba6e80-9e75-11e9-84cd-658cf1bb756e.png)

Reply:
7=>will create reduced volume
8,9,11,12,14,15,16 =>will create tpvv volume
17 to 25 =>Modified the code to handle the junk string as bad request

Reply:
Please find the attached truth table with the modified comments after the code modifications


![Truth table](https://user-images.githubusercontent.com/50358719/61039403-49f65300-a3ec-11e9-8bad-038a26718afc.png)